tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -2416,4 +2526,179 @@ public function listFileVersion(
 		$responseXml = HttpRequestHelper::getResponseXml($this->featureContext->getResponse());
 		return $responseXml->xpath("//d:response/d:href");
 	}
+
+	/**
+	 * return the etag for an element inside a space
+	 *
+	 * @param string $user requestor
+	 * @param string $space space name
+	 * @param string $path path to the file
+	 *
+	 * @return string
+	 * @throws GuzzleException
+	 */
+	public function userGetsEtagOfElementInASpace(string $user, string $space, string $path) {
+		$properties = '<?xml version="1.0"?>'
+			. '<d:propfind xmlns:d="DAV:" '
+			. 'xmlns:oc="http://owncloud.org/ns" '
+			. 'xmlns:ocs="http://open-collaboration-services.org/ns">'
+			.'<d:prop><d:getetag/></d:prop></d:propfind>';
+		$user = $this->featureContext->getActualUsername($user);
+		$space = $this->getSpaceByName($user, $space);
+
+		$fullUrl = $space['root']['webDavUrl'] . '/' . ltrim($path, '/');
+		$response = HttpRequestHelper::sendRequest(
+			$fullUrl,
+			$this->featureContext->getStepLineRef(),
+			'PROPFIND',
+			$user,
+			$this->featureContext->getPasswordForUser($user),
+			[],
+			$properties
+		);
+		$responseXml = HttpRequestHelper::getResponseXml($response);
+		$this->featureContext->setResponseXmlObject($responseXml);
+		return $this->featureContext->getEtagFromResponseXmlObject();
```

Comment:
can we use ` WebDavPropertiesContext::getCurrentEtagOfElement()` here and just give the correct path to it?

Reply:
due to the URL calculation, we cannot use that method. if we provide spaces as dav version, it will only look for the personal space and attaches the id of it to the URL but here we may need the URL for different spaces